src/controller/CHIPDevice.h

```diff
@@ -82,6 +82,7 @@ struct ControllerDeviceInitParams
     Inet::InetLayer * inetLayer                         = nullptr;
     PersistentStorageDelegate * storageDelegate         = nullptr;
     Credentials::OperationalCredentialSet * credentials = nullptr;
+    uint8_t * credentialsIndex                          = nullptr;
```

Comment:
why is this a pointer?

I would expect 'init params' to be input parameters for object initialization. Seeing a pointer inidicates to me either having this 'maybe not set (null)' in which case an 'optional' seems better or maybe because it is an output parameter, in which case I would prefer to place this separately to distinguish input vs output parameters

Reply:
Updated.